executor/restore.go

```diff
@@ -137,15 +137,13 @@ func (r *RestoreExecutor) buildRestoreObject(restore *k8upv1alpha1.Restore) (*ba
 		args = append(args, "-restoreSnap", restore.Spec.Snapshot)
 	}
 
-	methodDefined := false
-	if restore.Spec.RestoreMethod.Folder != nil {
+	switch {
+	case restore.Spec.RestoreMethod.Folder != nil:
 		args = append(args, "-restoreType", "folder")
-		methodDefined = true
-	}
-
-	if !methodDefined && restore.Spec.RestoreMethod.S3 != nil {
+	case restore.Spec.RestoreMethod.S3 != nil:
 		args = append(args, "-restoreType", "s3")
-		methodDefined = true
+	default:
+		r.Log.Error(err, "undefined restore method (-restoreType)", "namespace", r.Obj.GetMetaObject().GetNamespace())
```

Comment:
Should probably also be a condition.

Reply:
What would be the condition in your opinion? Since 

https://github.com/vshn/k8up/pull/204/files/3ac6b29a55917977ea2a16f3b6c2c0d741e52747#diff-7d65ba23b900330e9af0767e831c1a0ce44d9ceef8ec848084ac39c6d390e044R65

captures an error in here as a Condition anyway, I rather suggest to treat this as an error, i.e. to `return nil, err` (where err is a new error created with `fmt.Errorf` which replaces `r.Log.Error`.)

Reply:
Ah, I see. Yes, returning an error is the better option then.